src/shipchain_common/test_utils/assertions.py

```diff
@@ -0,0 +1,288 @@
+"""
+Copyright 2019 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import pytest
+from rest_framework import status
+
+# pylint: disable=too-many-branches
+# pylint: disable=too-many-arguments
+
+
+class EntityReferenceClass:
+    def __init__(self, resource=None, pk=None, attributes=None, relationships=None):
+        self.resource = resource
+        self.pk = pk
+        self.attributes = attributes
+        self.relationships = relationships
+
+    def __str__(self):
+        return f'Type: {self.resource}; ID: {self.pk}; ' \
+               f'attributes: {self.attributes}; relationships: {self.relationships}'
+
+
+def response_has_error(response, error):
+    if error is not None:
+        response_json = response.json()
+        assert error in response_json['errors'][0]['detail'], f'Error `{error}` not found in {response_json}'
```

Comment:
does it make sense to only check in the first error object? If there were multiple errors returned then would fail. It would be nice to have the option to check that either a single error was found, or that specific multiple errors were found.

Reply:
It might be useful to also add another check, `response_errors_equals`, or something like that, where you can assert that the errors in the response are equal to the list supplied, ensuring that there are no differences between the two. This could be useful in ensuring that only the errors you wanted were raised, and nothing more.